packages/zoe/src/zoeService/zoe.js

```diff
@@ -243,6 +218,68 @@ function makeZoe(vatAdminSvc, zcfBundleName = undefined) {
             zoeSeatAdmins.forEach(zoeSeatAdmin => zoeSeatAdmin.fail(reason));
           },
           stopAcceptingOffers: () => (acceptingOffers = false),
+          makeUserSeat: async (
+            invitationHandle,
+            initialAllocation,
+            proposal,
+          ) => {
+            const offerResultPromiseKit = makePromiseKit();
+            // Don't trigger Node.js's UnhandledPromiseRejectionWarning.
+            // This does not suppress any error messages.
+            offerResultPromiseKit.promise.catch(_ => {});
+            const exitObjPromiseKit = makePromiseKit();
+            // Don't trigger Node.js's UnhandledPromiseRejectionWarning.
+            // This does not suppress any error messages.
+            exitObjPromiseKit.promise.catch(_ => {});
+            const seatHandle = makeHandle('SeatHandle');
+
+            const { userSeat, notifier, zoeSeatAdmin } = makeZoeSeatAdminKit(
+              initialAllocation,
+              exitZoeSeatAdmin,
+              hasExited,
+              proposal,
+              withdrawPayments,
+              exitObjPromiseKit.promise,
+              offerResultPromiseKit.promise,
+            );
+
+            seatHandleToZoeSeatAdmin.init(seatHandle, zoeSeatAdmin);
+
+            const seatData = harden({ proposal, initialAllocation, notifier });
+
+            zoeSeatAdmins.add(zoeSeatAdmin);
+
+            E(/** @type {Promise<AddSeatObj>} */ (addSeatObjPromiseKit.promise))
```

Comment:
I do notice that `makePromiseKit` is typed as returning a `PromiseRecord<T>`. If we can provide the right `T` at the call site, perhaps we could avoid the casts at use sites like here. That would be better. But if it is not straightforward it should not delay this PR.

Reply:
That worked, thanks!